app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/activity/CafeteriaNotificationSettingsActivity.kt

```diff
@@ -0,0 +1,54 @@
+package de.tum.`in`.tumcampusapp.component.ui.cafeteria.activity
+
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.component.other.generic.activity.BaseActivity
+import de.tum.`in`.tumcampusapp.component.ui.cafeteria.CafeteriaNotificationSettings
+import de.tum.`in`.tumcampusapp.component.ui.cafeteria.CafeteriaNotificationSettingsAdapter
+import de.tum.`in`.tumcampusapp.component.ui.cafeteria.CafeteriaNotificationTime
+import kotlinx.android.synthetic.main.activity_cafeteria_notification_settings.*
+import org.joda.time.DateTime
+import org.joda.time.DateTimeConstants
+
+/**
+ * This activity enables the user to set a preferred notification time for a day of the week.
+ * The actual local storage of the preferences is done in the CafeteriaNotificationSettings class.
+ */
+class CafeteriaNotificationSettingsActivity : BaseActivity(R.layout.activity_cafeteria_notification_settings) {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val notificationSettings = CafeteriaNotificationSettings.getInstance(this)
+        val dailySchedule = buildDailySchedule(notificationSettings)
+
+        val layoutManager = LinearLayoutManager(this)
+        layoutManager.orientation = LinearLayoutManager.VERTICAL
+        notificationSettingsRecyclerView.layoutManager = layoutManager
+        notificationSettingsRecyclerView.setHasFixedSize(true)
+
+        val adapter = CafeteriaNotificationSettingsAdapter(this, dailySchedule)
+        notificationSettingsRecyclerView.adapter = adapter
+
+        notificationSettingsSaveButton.setOnClickListener {
+            notificationSettings.saveEntireSchedule(dailySchedule)
+            finish()
+        }
+    }
+
+    /**
+     * Reloads the settings into the dailySchedule list.
+     */
+    private fun buildDailySchedule(
+            settings: CafeteriaNotificationSettings): ArrayList<CafeteriaNotificationTime> {
+        return (DateTimeConstants.MONDAY until DateTimeConstants.SATURDAY)
+                .map {
+                    val day = DateTime.now().withDayOfWeek(it)
+                    val time = settings.retrieveLocalTime(day)
+                    CafeteriaNotificationTime(day, time)
+                }
+                .toCollection(ArrayList())
```

Comment:
It was mutable because list items were updated in `CafeteriaNotificationSettingsAdapter`. I’ve now changed the implementation so that the list can be immutable.